lib/src/chart/gauge_chart/gauge_chart_data.dart

```diff
@@ -0,0 +1,268 @@
+    return _LerpGaugeColor(a, b, t);
+  }
+}
+
+class _LerpGaugeColor implements GaugeColor, ColoredTicksGenerator {
+  _LerpGaugeColor(this.a, this.b, this.t);
+  final GaugeColor a;
+  final GaugeColor b;
+  final double t;
+
+  @override
+  Color getColor(double value) {
+    return Color.lerp(a.getColor(value), b.getColor(value), t)!;
+  }
+
+  @override
+  Iterable<ColoredTick> getColoredTicks() sync* {
+    if (a is ColoredTicksGenerator) {
+      for (final tick in (a as ColoredTicksGenerator).getColoredTicks()) {
+        yield ColoredTick(tick.position, tick.color.withOpacity(1 - t));
+      }
+    }
+    if (b is ColoredTicksGenerator) {
+      for (final tick in (b as ColoredTicksGenerator).getColoredTicks()) {
+        yield ColoredTick(tick.position, tick.color.withOpacity(t));
+      }
+    }
+  }
+}
+
+@immutable
+class SimpleGaugeColor with EquatableMixin implements GaugeColor {
+  const SimpleGaugeColor({required this.color});
+  final Color color;
+
+  @override
+  Color getColor(double value) => color;
+
+  @override
+  List<Object?> get props => [color];
+}
+
+class VariableGaugeColor
+    with EquatableMixin
+    implements GaugeColor, ColoredTicksGenerator {
+  VariableGaugeColor({
+    required this.limits,
+    required this.colors,
+  })  : assert(
+          colors.length - 1 == limits.length,
+          'length of limits should be equals to colors length minus one',
+        ),
+        assert(
+          limits.length <= 1 || limits.reduce((a, b) => a < b ? 0 : 2) == 0,
+          'the limits list should be sorted in ascending order',
+        ),
+        assert(
+          limits.first > 0 || limits.last < 1.0,
+          'limits values should be in range 0, 1 (exclusive)',
+        );
+
+  final List<double> limits;
+  final List<Color> colors;
+
+  @override
+  Color getColor(double value) {
+    for (var i = 0; i < limits.length; i++) {
+      if (value < limits[i]) return colors[i];
+    }
+    return colors.last;
+  }
+
+  @override
+  Iterable<ColoredTick> getColoredTicks() sync* {
+    for (var i = 0; i < limits.length; i++) {
+      yield ColoredTick(limits[i], colors[i + 1]);
+    }
+  }
+
+  @override
+  List<Object?> get props => [limits, colors];
+}
+
+enum GaugeTickPosition {
+  inner,
+  outer,
+  center,
+}
+
+@immutable
+class GaugeTicks with EquatableMixin {
+  const GaugeTicks({
+    this.count = 3,
+    this.radius = 3.0,
+    required this.color,
+    this.position = GaugeTickPosition.outer,
+    this.margin = 3,
+    this.showChangingColorTicks = true,
+  })  : assert(count > 2, 'count should be >= 2'),
+        assert(radius > 0, 'radius should be > 0');
+  final int count;
+  final double radius;
+  final Color color;
+  final GaugeTickPosition position;
+  final double margin;
+  final bool showChangingColorTicks;
+
+  static GaugeTicks? lerp(GaugeTicks? a, GaugeTicks? b, double t) {
+    // TODO(FlorianArnould): if showChangingColorTicks are different
```

Comment:
Is this to do necessary?

Reply:
Yes if we want a smooth transition on the color ticks
But would be deleted if we go for another approach like discussed here 
https://github.com/imaNNeo/fl_chart/pull/1424#discussion_r1326595223